lib/honeybadger/agent.rb

```diff
@@ -122,6 +122,8 @@ def notify(exception_or_opts, opts = {})
 
       notice = Notice.new(config, opts)
 
+      config.ruby[:before_notify].each { |hook| hook.call(notice) } if config.ruby[:before_notify]
```

Comment:
Here is where we would want to be very careful about catching any errors that could be raised by a "before notify" hook. Do we have any thoughts on how to be precise about this? Or about how we would want to squelch or report such errors?

Also, the `config.ruby[:before_notify]` feels funky to me. Is there a better way to access them at this point?

Reply:
We could potentially also apply the filters in `Notice#initialize`, since `Notice` also handles applying the other callbacks. The `config` object is available inside `Notice`. What do you guys think?

Reply:
I'm not sure I like the idea of the notice being the one to run the hooks. That seems like a responsibility that shouldn't fall on it.

Reply:
Yeah, that makes sense. I would like to simplify `Notice` in the future, or create a simpler object that is passed to callbacks. `Notice` currently encapsulates a lot of private logic which the user shouldn't come in contact with. Ideally the object which is passed to callbacks would be a simple class with a handful of `attr_accessor`s for modifying data. Small steps, though. :)